sklearn/utils/testing.py

```diff
@@ -625,8 +625,9 @@ def run_test(*args, **kwargs):
             import matplotlib
             matplotlib.use('Agg', warn=False)
             # this fails if no $DISPLAY specified
```

Comment:
As I said in one of my comment below, I am not really sure whether we want to keep the plt.figure() line so hold off changing the comment for now.